### ISSUE TITLE
How many machines needed for OpenStack MAAS Getting Insufficient Peers

### ISSUE THREAD

**Issue by qgriffith** [conjure-up/conjure-up#490]
Ubuntu 16.04 using MAAS **Version 2.0.0+bzr5189-0ubuntu1 (16.04.1)** JUJU: 2.0-rc3-0ubuntu1~16.04.1~juju1 conjure: **2.0.1-0~201610061938~ubuntu16.04.**

I have five blades managed by MAAS. When I conjure-up openstack and select the MAAS option, the job never completes it appears to be waiting on resources. I get Insufficient Peers.  The description for Openstack Base on MAAS indicates you only need four machines in MAAS.

It looks like the first server that conjure-up provisions does not get any lxc containers on it. The other four servers I have get only three lxc containers on them, and then everything gets into a wait state.

When I reduce the amount of units before deploying, Ceph Mon hangs because it expects to have three of them, however I reduced it to one. It seems like reducing Ceph Mon won't work because something is hard coded to have to have three.

**Comment by battlemidget** [comment]
So with Juju a node is always allocated as the admin/controller node which means at least 1 system will be reserved just for that. So essentially you would only have 4 systems at your disposal to deploy OpenStack. With that said 5 machines should be enough for you to deploy with; can you provide me with `juju status --format yaml`?

**Comment by qgriffith** [comment]
I just kicked off a new run with one compute unit, and one ceph osd unit, in hopes it will work. I will send you the output once it gets to the part it hangs. Thanks so much

**Comment by qgriffith** [comment]
As soon as each blade got three lxc containers it pretty much stops with the following

![screen shot 2016-10-12 at 8 41 11 am](https://cloud.githubusercontent.com/assets/203675/19310370/ba2fb89c-9057-11e6-8967-ad2e1d86d188.png)


```
model:
  name: conjure-up
  controller: perch
  cloud: maas
  version: 2.0-rc3
machines:
  "0":
    juju-status:
      current: started
      since: 12 Oct 2016 07:32:24-05:00
      version: 2.0-rc3
    dns-name: 10.5.100.52
    instance-id: 4y3h7r
    machine-status:
      current: running
      message: Deployed
      since: 12 Oct 2016 07:31:39-05:00
    series: xenial
    containers:
      0/lxd/0:
        juju-status:
          current: started
          since: 12 Oct 2016 07:35:45-05:00
          version: 2.0-rc3
        dns-name: 10.5.100.124
        instance-id: juju-685564-0-lxd-0
        machine-status:
          current: running
          message: Container started
          since: 12 Oct 2016 07:35:01-05:00
        series: xenial
      0/lxd/1:
        juju-status:
          current: started
          since: 12 Oct 2016 07:35:43-05:00
          version: 2.0-rc3
        dns-name: 10.5.100.128
        instance-id: juju-685564-0-lxd-1
        machine-status:
          current: running
          message: Container started
          since: 12 Oct 2016 07:35:11-05:00
        series: xenial
      0/lxd/2:
        juju-status:
          current: started
          since: 12 Oct 2016 07:35:59-05:00
          version: 2.0-rc3
        dns-name: 10.5.100.131
        instance-id: juju-685564-0-lxd-2
        machine-status:
          current: running
          message: Container started
          since: 12 Oct 2016 07:35:21-05:00
        series: xenial
    hardware: arch=amd64 cores=24 mem=32768M availability-zone=default
  "1":
    juju-status:
      current: started
      since: 12 Oct 2016 07:32:12-05:00
      version: 2.0-rc3
    dns-name: 10.5.100.53
    instance-id: 4y3h7s
    machine-status:
      current: running
      message: Deployed
      since: 12 Oct 2016 07:31:34-05:00
    series: xenial
    containers:
      1/lxd/0:
        juju-status:
          current: started
          since: 12 Oct 2016 07:35:44-05:00
          version: 2.0-rc3
        dns-name: 10.5.100.123
        instance-id: juju-685564-1-lxd-0
        machine-status:
          current: running
          message: Container started
          since: 12 Oct 2016 07:35:11-05:00
        series: xenial
      1/lxd/1:
        juju-status:
          current: started
          since: 12 Oct 2016 07:35:58-05:00
          version: 2.0-rc3
        dns-name: 10.5.100.130
        instance-id: juju-685564-1-lxd-1
        machine-status:
          current: running
          message: Container started
          since: 12 Oct 2016 07:35:20-05:00
        series: xenial
      1/lxd/2:
        juju-status:
          current: started
          since: 12 Oct 2016 07:36:03-05:00
          version: 2.0-rc3
        dns-name: 10.5.100.132
        instance-id: juju-685564-1-lxd-2
        machine-status:
          current: running
          message: Container started
          since: 12 Oct 2016 07:35:31-05:00
        series: xenial
    hardware: arch=amd64 cores=24 mem=32768M availability-zone=default
  "2":
    juju-status:
      current: started
      since: 12 Oct 2016 07:32:09-05:00
      version: 2.0-rc3
    dns-name: 10.5.100.56
    instance-id: 4y3h7t
    machine-status:
      current: running
      message: Deployed
      since: 12 Oct 2016 07:31:22-05:00
    series: xenial
    containers:
      2/lxd/0:
        juju-status:
          current: started
          since: 12 Oct 2016 07:35:30-05:00
          version: 2.0-rc3
        dns-name: 10.5.100.121
        instance-id: juju-685564-2-lxd-0
        machine-status:
          current: running
          message: Container started
          since: 12 Oct 2016 07:34:59-05:00
        series: xenial
      2/lxd/1:
        juju-status:
          current: started
          since: 12 Oct 2016 07:35:38-05:00
          version: 2.0-rc3
        dns-name: 10.5.100.127
        instance-id: juju-685564-2-lxd-1
        machine-status:
          current: running
          message: Container started
          since: 12 Oct 2016 07:35:08-05:00
        series: xenial
      2/lxd/2:
        juju-status:
          current: started
          since: 12 Oct 2016 07:35:50-05:00
          version: 2.0-rc3
        dns-name: 10.5.100.129
        instance-id: juju-685564-2-lxd-2
        machine-status:
          current: running
          message: Container started
          since: 12 Oct 2016 07:35:19-05:00
        series: xenial
    hardware: arch=amd64 cores=24 mem=32768M availability-zone=default
  "3":
    juju-status:
      current: started
      since: 12 Oct 2016 07:32:14-05:00
      version: 2.0-rc3
    dns-name: 10.5.100.51
    instance-id: 4y3h7q
    machine-status:
      current: running
      message: Deployed
      since: 12 Oct 2016 07:31:34-05:00
    series: xenial
    containers:
      3/lxd/0:
        juju-status:
          current: pending
          since: 12 Oct 2016 07:22:12-05:00
        instance-id: juju-685564-3-lxd-0
        machine-status:
          current: running
          message: Container started
          since: 12 Oct 2016 07:34:54-05:00
        series: xenial
      3/lxd/1:
        juju-status:
          current: pending
          since: 12 Oct 2016 07:22:22-05:00
        instance-id: juju-685564-3-lxd-1
        machine-status:
          current: running
          message: Container started
          since: 12 Oct 2016 07:35:03-05:00
        series: xenial
      3/lxd/2:
        juju-status:
          current: pending
          since: 12 Oct 2016 07:22:47-05:00
        instance-id: juju-685564-3-lxd-2
        machine-status:
          current: running
          message: Container started
          since: 12 Oct 2016 07:34:45-05:00
        series: xenial
    hardware: arch=amd64 cores=24 mem=32768M availability-zone=default
applications:
  ceph-mon:
    charm: cs:xenial/ceph-mon-3
    series: xenial
    os: ubuntu
    charm-origin: jujucharms
    charm-name: ceph-mon
    charm-rev: 3
    exposed: false
    application-status:
      current: blocked
      message: Insufficient peer units to bootstrap cluster (require 3)
      since: 12 Oct 2016 07:40:43-05:00
    relations:
      client:
      - cinder-ceph
      - glance
      - nova-compute
      mon:
      - ceph-mon
      osd:
      - ceph-osd
      radosgw:
      - ceph-radosgw
    units:
      ceph-mon/0:
        workload-status:
          current: blocked
          message: Insufficient peer units to bootstrap cluster (require 3)
          since: 12 Oct 2016 07:40:43-05:00
        juju-status:
          current: idle
          since: 12 Oct 2016 07:40:44-05:00
          version: 2.0-rc3
        machine: 1/lxd/0
        public-address: 10.5.100.123
      ceph-mon/1:
        workload-status:
          current: blocked
          message: Insufficient peer units to bootstrap cluster (require 3)
          since: 12 Oct 2016 07:40:43-05:00
        juju-status:
          current: idle
          since: 12 Oct 2016 07:40:43-05:00
          version: 2.0-rc3
        leader: true
        machine: 2/lxd/0
        public-address: 10.5.100.121
      ceph-mon/2:
        workload-status:
          current: waiting
          message: waiting for machine
          since: 12 Oct 2016 07:22:08-05:00
        juju-status:
          current: allocating
          since: 12 Oct 2016 07:22:08-05:00
        machine: 3/lxd/0
  ceph-osd:
    charm: cs:xenial/ceph-osd-3
    series: xenial
    os: ubuntu
    charm-origin: jujucharms
    charm-name: ceph-osd
    charm-rev: 3
    exposed: false
    application-status:
      current: waiting
      message: 'Incomplete relation: monitor'
      since: 12 Oct 2016 07:39:17-05:00
    relations:
      mon:
      - ceph-mon
    units:
      ceph-osd/0:
        workload-status:
          current: waiting
          message: 'Incomplete relation: monitor'
          since: 12 Oct 2016 07:39:17-05:00
        juju-status:
          current: idle
          since: 12 Oct 2016 07:39:17-05:00
          version: 2.0-rc3
        leader: true
        machine: "1"
        public-address: 10.5.100.53
  ceph-radosgw:
    charm: cs:xenial/ceph-radosgw-5
    series: xenial
    os: ubuntu
    charm-origin: jujucharms
    charm-name: ceph-radosgw
    charm-rev: 5
    exposed: false
    application-status:
      current: blocked
      message: 'Missing relations: mon; incomplete relations: identity'
      since: 12 Oct 2016 07:40:18-05:00
    relations:
      cluster:
      - ceph-radosgw
      identity-service:
      - keystone
      mon:
      - ceph-mon
    units:
      ceph-radosgw/0:
        workload-status:
          current: blocked
          message: 'Missing relations: mon; incomplete relations: identity'
          since: 12 Oct 2016 07:40:18-05:00
        juju-status:
          current: idle
          since: 12 Oct 2016 07:40:18-05:00
          version: 2.0-rc3
        leader: true
        machine: 0/lxd/0
        public-address: 10.5.100.124
  cinder:
    charm: cs:xenial/cinder-2
    series: xenial
    os: ubuntu
    charm-origin: jujucharms
    charm-name: cinder
    charm-rev: 2
    exposed: false
    application-status:
      current: waiting
      message: 'Incomplete relations: messaging, identity'
      since: 12 Oct 2016 07:42:26-05:00
    relations:
      amqp:
      - rabbitmq-server
      cinder-volume-service:
      - nova-cloud-controller
      cluster:
      - cinder
      identity-service:
      - keystone
      image-service:
      - glance
      shared-db:
      - mysql
      storage-backend:
      - cinder-ceph
    units:
      cinder/0:
        workload-status:
          current: waiting
          message: 'Incomplete relations: messaging, identity'
          since: 12 Oct 2016 07:42:26-05:00
        juju-status:
          current: executing
          message: running amqp-relation-changed hook
          since: 12 Oct 2016 07:42:26-05:00
          version: 2.0-rc3
        leader: true
        machine: 1/lxd/1
        open-ports:
        - 8776/tcp
        public-address: 10.5.100.130
        subordinates:
          cinder-ceph/0:
            workload-status:
              current: waiting
              message: agent initializing
              since: 12 Oct 2016 07:41:23-05:00
            juju-status:
              current: allocating
              since: 12 Oct 2016 07:41:19-05:00
              version: 2.0-rc3
            leader: true
            upgrading-from: cs:xenial/cinder-ceph-2
            public-address: 10.5.100.130
  cinder-ceph:
    charm: cs:xenial/cinder-ceph-2
    series: xenial
    os: ubuntu
    charm-origin: jujucharms
    charm-name: cinder-ceph
    charm-rev: 2
    exposed: false
    application-status:
      current: waiting
      message: agent initializing
      since: 12 Oct 2016 07:41:23-05:00
    relations:
      ceph:
      - ceph-mon
      storage-backend:
      - cinder
    subordinate-to:
    - cinder
  glance:
    charm: cs:xenial/glance-2
    series: xenial
    os: ubuntu
    charm-origin: jujucharms
    charm-name: glance
    charm-rev: 2
    exposed: false
    application-status:
      current: waiting
      message: 'Incomplete relations: messaging, storage-backend, identity'
      since: 12 Oct 2016 07:42:22-05:00
    relations:
      amqp:
      - rabbitmq-server
      ceph:
      - ceph-mon
      cluster:
      - glance
      identity-service:
      - keystone
      image-service:
      - cinder
      - nova-cloud-controller
      - nova-compute
      shared-db:
      - mysql
    units:
      glance/0:
        workload-status:
          current: waiting
          message: 'Incomplete relations: messaging, storage-backend, identity'
          since: 12 Oct 2016 07:42:22-05:00
        juju-status:
          current: executing
          message: running image-service-relation-joined hook
          since: 12 Oct 2016 07:42:22-05:00
          version: 2.0-rc3
        leader: true
        machine: 2/lxd/1
        open-ports:
        - 9292/tcp
        public-address: 10.5.100.127
  keystone:
    charm: cs:xenial/keystone-2
    series: xenial
    os: ubuntu
    charm-origin: jujucharms
    charm-name: keystone
    charm-rev: 2
    exposed: false
    application-status:
      current: waiting
      message: waiting for machine
      since: 12 Oct 2016 07:22:21-05:00
    relations:
      cluster:
      - keystone
      identity-service:
      - ceph-radosgw
      - cinder
      - glance
      - neutron-api
      - nova-cloud-controller
      - openstack-dashboard
      shared-db:
      - mysql
    units:
      keystone/0:
        workload-status:
          current: waiting
          message: waiting for machine
          since: 12 Oct 2016 07:22:21-05:00
        juju-status:
          current: allocating
          since: 12 Oct 2016 07:22:21-05:00
        machine: 3/lxd/1
  mysql:
    charm: cs:xenial/percona-cluster-2
    series: xenial
    os: ubuntu
    charm-origin: jujucharms
    charm-name: percona-cluster
    charm-rev: 2
    exposed: false
    application-status:
      current: active
      message: Unit is ready
      since: 12 Oct 2016 07:42:00-05:00
    relations:
      cluster:
      - mysql
      shared-db:
      - cinder
      - glance
      - keystone
      - neutron-api
      - nova-cloud-controller
    units:
      mysql/0:
        workload-status:
          current: active
          message: Unit is ready
          since: 12 Oct 2016 07:42:00-05:00
        juju-status:
          current: idle
          since: 12 Oct 2016 07:42:00-05:00
          version: 2.0-rc3
        leader: true
        machine: 0/lxd/1
        public-address: 10.5.100.128
  neutron-api:
    charm: cs:xenial/neutron-api-3
    series: xenial
    os: ubuntu
    charm-origin: jujucharms
    charm-name: neutron-api
    charm-rev: 3
    exposed: false
    application-status:
      current: waiting
      message: 'Incomplete relations: messaging, identity'
      since: 12 Oct 2016 07:42:30-05:00
    relations:
      amqp:
      - rabbitmq-server
      cluster:
      - neutron-api
      identity-service:
      - keystone
      neutron-api:
      - nova-cloud-controller
      neutron-plugin-api:
      - neutron-gateway
      - neutron-openvswitch
      shared-db:
      - mysql
    units:
      neutron-api/0:
        workload-status:
          current: waiting
          message: 'Incomplete relations: messaging, identity'
          since: 12 Oct 2016 07:42:30-05:00
        juju-status:
          current: executing
          message: running amqp-relation-changed hook
          since: 12 Oct 2016 07:42:18-05:00
          version: 2.0-rc3
        leader: true
        machine: 1/lxd/2
        open-ports:
        - 9696/tcp
        public-address: 10.5.100.132
  neutron-gateway:
    charm: cs:xenial/neutron-gateway-2
    series: xenial
    os: ubuntu
    charm-origin: jujucharms
    charm-name: neutron-gateway
    charm-rev: 2
    exposed: false
    application-status:
      current: waiting
      message: 'Incomplete relations: messaging'
      since: 12 Oct 2016 07:42:14-05:00
    relations:
      amqp:
      - rabbitmq-server
      cluster:
      - neutron-gateway
      juju-info:
      - ntp
      neutron-plugin-api:
      - neutron-api
      quantum-network-service:
      - nova-cloud-controller
    units:
      neutron-gateway/0:
        workload-status:
          current: waiting
          message: 'Incomplete relations: messaging'
          since: 12 Oct 2016 07:42:14-05:00
        juju-status:
          current: executing
          message: running amqp-relation-changed hook
          since: 12 Oct 2016 07:42:14-05:00
          version: 2.0-rc3
        leader: true
        machine: "0"
        public-address: 10.5.100.52
        subordinates:
          ntp/1:
            workload-status:
              current: unknown
              since: 12 Oct 2016 07:37:12-05:00
            juju-status:
              current: idle
              since: 12 Oct 2016 07:37:14-05:00
              version: 2.0-rc3
            upgrading-from: cs:xenial/ntp-0
            public-address: 10.5.100.52
  neutron-openvswitch:
    charm: cs:xenial/neutron-openvswitch-3
    series: xenial
    os: ubuntu
    charm-origin: jujucharms
    charm-name: neutron-openvswitch
    charm-rev: 3
    exposed: false
    application-status:
      current: waiting
      message: 'Incomplete relations: messaging'
      since: 12 Oct 2016 07:41:28-05:00
    relations:
      amqp:
      - rabbitmq-server
      neutron-plugin:
      - nova-compute
      neutron-plugin-api:
      - neutron-api
    subordinate-to:
    - nova-compute
  nova-cloud-controller:
    charm: cs:xenial/nova-cloud-controller-3
    series: xenial
    os: ubuntu
    charm-origin: jujucharms
    charm-name: nova-cloud-controller
    charm-rev: 3
    exposed: false
    application-status:
      current: waiting
      message: 'Incomplete relations: neutron-api, image, database, identity, messaging'
      since: 12 Oct 2016 07:42:05-05:00
    relations:
      amqp:
      - rabbitmq-server
      cinder-volume-service:
      - cinder
      cloud-compute:
      - nova-compute
      cluster:
      - nova-cloud-controller
      identity-service:
      - keystone
      image-service:
      - glance
      neutron-api:
      - neutron-api
      quantum-network-service:
      - neutron-gateway
      shared-db:
      - mysql
    units:
      nova-cloud-controller/0:
        workload-status:
          current: waiting
          message: 'Incomplete relations: neutron-api, image, database, identity,
            messaging'
          since: 12 Oct 2016 07:42:05-05:00
        juju-status:
          current: executing
          message: running cinder-volume-service-relation-joined hook
          since: 12 Oct 2016 07:42:05-05:00
          version: 2.0-rc3
        leader: true
        machine: 2/lxd/2
        open-ports:
        - 8774/tcp
        public-address: 10.5.100.129
  nova-compute:
    charm: cs:xenial/nova-compute-2
    series: xenial
    os: ubuntu
    charm-origin: jujucharms
    charm-name: nova-compute
    charm-rev: 2
    exposed: false
    application-status:
      current: waiting
      message: 'Incomplete relations: storage-backend'
      since: 12 Oct 2016 07:42:18-05:00
    relations:
      amqp:
      - rabbitmq-server
      ceph:
      - ceph-mon
      cloud-compute:
      - nova-cloud-controller
      compute-peer:
      - nova-compute
      image-service:
      - glance
      juju-info:
      - ntp
      neutron-plugin:
      - neutron-openvswitch
    units:
      nova-compute/0:
        workload-status:
          current: waiting
          message: 'Incomplete relations: storage-backend'
          since: 12 Oct 2016 07:42:18-05:00
        juju-status:
          current: executing
          message: running cloud-compute-relation-changed hook
          since: 12 Oct 2016 07:42:18-05:00
          version: 2.0-rc3
        leader: true
        machine: "1"
        public-address: 10.5.100.53
        subordinates:
          neutron-openvswitch/0:
            workload-status:
              current: waiting
              message: 'Incomplete relations: messaging'
              since: 12 Oct 2016 07:41:28-05:00
            juju-status:
              current: idle
              since: 12 Oct 2016 07:41:28-05:00
              version: 2.0-rc3
            leader: true
            upgrading-from: cs:xenial/neutron-openvswitch-3
            public-address: 10.5.100.53
          ntp/0:
            workload-status:
              current: unknown
              since: 12 Oct 2016 07:37:09-05:00
            juju-status:
              current: idle
              since: 12 Oct 2016 07:39:18-05:00
              version: 2.0-rc3
            leader: true
            upgrading-from: cs:xenial/ntp-0
            public-address: 10.5.100.53
  ntp:
    charm: cs:xenial/ntp-0
    series: xenial
    os: ubuntu
    charm-origin: jujucharms
    charm-name: ntp
    charm-rev: 0
    exposed: false
    application-status:
      current: unknown
      since: 12 Oct 2016 07:37:09-05:00
    relations:
      juju-info:
      - neutron-gateway
      - nova-compute
      ntp-peers:
      - ntp
    subordinate-to:
    - neutron-gateway
    - nova-compute
  openstack-dashboard:
    charm: cs:xenial/openstack-dashboard-2
    series: xenial
    os: ubuntu
    charm-origin: jujucharms
    charm-name: openstack-dashboard
    charm-rev: 2
    exposed: false
    application-status:
      current: waiting
      message: waiting for machine
      since: 12 Oct 2016 07:22:46-05:00
    relations:
      cluster:
      - openstack-dashboard
      identity-service:
      - keystone
    units:
      openstack-dashboard/0:
        workload-status:
          current: waiting
          message: waiting for machine
          since: 12 Oct 2016 07:22:46-05:00
        juju-status:
          current: allocating
          since: 12 Oct 2016 07:22:46-05:00
        machine: 3/lxd/2
  rabbitmq-server:
    charm: cs:xenial/rabbitmq-server-5
    series: xenial
    os: ubuntu
    charm-origin: jujucharms
    charm-name: rabbitmq-server
    charm-rev: 5
    exposed: false
    application-status:
      current: active
      message: Unit is ready
      since: 12 Oct 2016 07:42:27-05:00
    relations:
      amqp:
      - cinder
      - glance
      - neutron-api
      - neutron-gateway
      - neutron-openvswitch
      - nova-cloud-controller
      - nova-compute
      cluster:
      - rabbitmq-server
    units:
      rabbitmq-server/0:
        workload-status:
          current: active
          message: Unit is ready
          since: 12 Oct 2016 07:42:27-05:00
        juju-status:
          current: executing
          message: running amqp-relation-changed hook
          since: 12 Oct 2016 07:42:06-05:00
          version: 2.0-rc3
        leader: true
        machine: 0/lxd/2
        open-ports:
        - 5672/tcp
        public-address: 10.5.100.131
```

**Comment by battlemidget** [comment]
Does running `juju debug-log` give you any indication it's still installing packages or doing any other work? Also nice hourglass icon that from a macOS terminal? :)

**Comment by battlemidget** [comment]
also join #ubuntu-solutions on freenode.net so we can talk more and get this sorted

**Comment by qgriffith** [comment]
It looks like it is doing some stuff with the nodes it has created

```
unit-nova-compute-0: 07:47:53 INFO unit.nova-compute/0.juju-log cloud-compute:25: Wrote template /etc/apparmor.d/usr.bin.nova-compute.
unit-cinder-ceph-0: 07:47:53 INFO juju.worker.leadership cinder-ceph/0 will renew cinder-ceph leadership at 2016-10-12 12:48:23.621008615 +0000 UTC
unit-nova-compute-0: 07:47:54 INFO unit.nova-compute/0.juju-log cloud-compute:25: Loaded template from templates/libvirtd.conf
unit-nova-compute-0: 07:47:54 INFO unit.nova-compute/0.juju-log cloud-compute:25: Rendering from template: libvirtd.conf
unit-nova-compute-0: 07:47:54 INFO unit.nova-compute/0.juju-log cloud-compute:25: Wrote template /etc/libvirt/libvirtd.conf.
unit-nova-compute-0: 07:47:54 DEBUG unit.nova-compute/0.juju-log cloud-compute:25: Generating template context for ceph
unit-nova-compute-0: 07:47:55 INFO unit.nova-compute/0.juju-log cloud-compute:25: Missing required data: key auth
unit-nova-compute-0: 07:47:55 INFO unit.nova-compute/0.juju-log cloud-compute:25: Loaded template from /var/lib/juju/agents/unit-nova-compute-0/charm/hooks/charmhelpers/contrib/openstack/templates/ceph.conf
unit-nova-compute-0: 07:47:55 INFO unit.nova-compute/0.juju-log cloud-compute:25: Rendering from template: ceph.conf
unit-nova-compute-0: 07:47:55 INFO unit.nova-compute/0.juju-log cloud-compute:25: Wrote template /var/lib/charm/nova-compute/ceph.conf.
unit-nova-compute-0: 07:47:55 DEBUG unit.nova-compute/0.juju-log cloud-compute:25: Generating template context for ceph
unit-nova-compute-0: 07:47:55 INFO unit.nova-compute/0.juju-log cloud-compute:25: Missing required data: key auth
unit-nova-compute-0: 07:47:55 INFO unit.nova-compute/0.juju-log cloud-compute:25: Loaded template from templates/secret.xml
unit-nova-compute-0: 07:47:55 INFO unit.nova-compute/0.juju-log cloud-compute:25: Rendering from template: secret.xml
unit-nova-compute-0: 07:47:55 INFO unit.nova-compute/0.juju-log cloud-compute:25: Wrote template /etc/ceph/secret.xml.
unit-nova-compute-0: 07:47:55 INFO unit.nova-compute/0.juju-log cloud-compute:25: Loaded template from templates/qemu.conf
unit-nova-compute-0: 07:47:55 INFO unit.nova-compute/0.juju-log cloud-compute:25: Rendering from template: qemu.conf
unit-nova-compute-0: 07:47:55 INFO unit.nova-compute/0.juju-log cloud-compute:25: Wrote template /etc/libvirt/qemu.conf.
unit-nova-compute-0: 07:47:56 INFO unit.nova-compute/0.juju-log cloud-compute:25: Loaded template from templates/libvirt-bin.override
unit-nova-compute-0: 07:47:56 INFO unit.nova-compute/0.juju-log cloud-compute:25: Rendering from template: libvirt-bin.override
unit-nova-compute-0: 07:47:56 INFO unit.nova-compute/0.juju-log cloud-compute:25: Wrote template /etc/init/libvirt-bin.override.
unit-nova-compute-0: 07:47:56 DEBUG unit.nova-compute/0.juju-log cloud-compute:25: Generating template context for amqp
unit-neutron-gateway-0: 07:47:56 INFO juju.worker.leadership neutron-gateway/0 will renew neutron-gateway leadership at 2016-10-12 12:48:26.429326476 +0000 UTC
unit-nova-compute-0: 07:47:56 DEBUG unit.nova-compute/0.juju-log cloud-compute:25: Generating template context for image-service.
unit-neutron-openvswitch-0: 07:47:57 INFO juju.worker.uniter no operations in progress; waiting for changes
unit-nova-compute-0: 07:47:57 INFO unit.nova-compute/0.juju-log cloud-compute:25: Missing required relation settings for Quantum: neutron_admin_username auth_port neutron_admin_password keystone_host neutron_admin_tenant_name
unit-nova-compute-0: 07:47:57 INFO unit.nova-compute/0.juju-log cloud-compute:25: Generated config context for neutron network manager.
unit-nova-compute-0: 07:47:58 DEBUG unit.nova-compute/0.juju-log cloud-compute:25: Generating template context for ceph
unit-nova-compute-0: 07:47:58 INFO unit.nova-compute/0.juju-log cloud-compute:25: Missing required data: key auth
unit-nova-compute-0: 07:47:58 DEBUG unit.nova-compute/0.juju-log cloud-compute:25: 0 section(s) found
unit-nova-compute-0: 07:47:59 INFO unit.nova-compute/0.juju-log cloud-compute:25: Loaded template from templates/mitaka/nova.conf
unit-nova-compute-0: 07:47:59 INFO unit.nova-compute/0.juju-log cloud-compute:25: Rendering from template: nova.conf
unit-nova-compute-0: 07:47:59 INFO unit.nova-compute/0.juju-log cloud-compute:25: Wrote template /etc/nova/nova.conf.
unit-nova-compute-0: 07:47:59 INFO unit.nova-compute/0.juju-log cloud-compute:25: Loaded template from templates/libvirt-bin
unit-nova-compute-0: 07:47:59 INFO unit.nova-compute/0.juju-log cloud-compute:25: Rendering from template: libvirt-bin
```

**Comment by qgriffith** [comment]
Joined the channel. Oh yes it is Mac :)

**Comment by qgriffith** [comment]
The problem was firmware on a giving blade. The lxc containers that spin up would get an IP but I couldn't not ping them. When I attached to the containers I noticed they didn't have a lxdbr0 interface like the working ones did.  I switched out the blade to one with more recent firmware and the build finished. So key note always make sure your firmware is up to date on hardware before spinning up openstack.
